metadataclient/mds.py

```diff
@@ -150,38 +149,72 @@ def run_stop_given_uid(self, uid):
         params = self.queryfactory(query={'uid': uid},
                                    signature='run_start_given_uid')
         response = self._get(self._rstop_url, params=params)
-        return self._cache_run_stop(run_stop=response,
+        return self._cache_run_stop(response,
                                     self._RUN_STOP_CACHE)
 
     def descriptor_given_uid(self, uid):
         uid = self.doc_or_uid_to_uid(uid)
         try:
-            return self.DESCRIPTOR_CACHE[uid]
+            return self._DESCRIPTOR_CACHE[uid]
         except KeyError:
             pass
         params = self.queryfactory(query={'uid': uid},
                                    signature='run_start_given_uid')
         response = self._get(self._desc_url, params=params)
-        return self._cache_descriptor(descriptor=response,
+        return self._cache_descriptor(response,
                                       self._DESCRIPTOR_CACHE)
 
-    def descriptors_by_start(run_start):
+    def descriptors_by_start(self, run_start):
         rstart_uid = self.doc_or_uid_to_uid(run_start)
-        params = self.queryfactor(query={'run_start': rstart_uid},
+        params = self.queryfactory(query={'run_start': rstart_uid},
                              signature='descriptors_by_start')
         self._get(self._desc_url, params=params)
-        return self._cache_descriptor(descriptor=response,
+        return self._cache_descriptor(response,
                                       self._DESCRIPTOR_CACHE)
 
     def stop_by_start(self, run_start):
-        uid = self.doc_or_uid_to_uid()
+        uid = self.doc_or_uid_to_uid(run_start)
         params = self.queryfactory(query={'run_start': uid},
                                    signature='stop_by_start')
         response = self._get(self._rstop_url, params=params)
         return self._cache_run_stop(response, self._RUN_STOP_CACHE)
 
-    def get_events_generator(descriptor, convert_arrays=True):
-        pass
+    def get_events_generator(self, descriptor, convert_arrays=True):
+       descriptor_uid = self.doc_or_uid_to_uid(descriptor)
+       descriptor = self.descriptor_given_uid(descriptor_uid)
+       params = self.queryfactory(query={'descriptor': descriptor,
+                                         'convert_arrays': convert_arrays},
+                                  signature='get_events_generator')
+       events = self._get(self._event_url, params=params)
+       yield events
+
+    def _transpose(self, in_data, keys, field):
```

Comment:
This was for transposing events from the days we stored event data and timestamp as tuples. I have removed this in my next commit